src/routes.jsx

```diff
@@ -0,0 +1,70 @@
+import { Navigate } from 'react-router';
+
+import { EmbeddedRegistrationRoute, NotFoundPage, UnAuthOnlyRoute } from './common-components';
+import { LOGIN_PAGE, PAGE_NOT_FOUND } from './data/constants';
+import { ForgotPasswordPage } from './forgot-password';
+import Logistration from './logistration/Logistration';
+import { ProgressiveProfiling } from './progressive-profiling';
+import { RegistrationPage } from './register';
+import { ResetPasswordPage } from './reset-password';
+
+import Main from './Main';
+
+const routes = [
+  {
+    id: 'org.openedx.frontend.route.authn.main',
+    Component: Main,
+    children: [
+      {
+        path: 'register-embedded',
+        element: (
+          <EmbeddedRegistrationRoute><RegistrationPage /></EmbeddedRegistrationRoute>
+        )
+      },
+      {
+        path: 'login',
+        element: (
+          <UnAuthOnlyRoute><Logistration selectedPage={LOGIN_PAGE} /></UnAuthOnlyRoute>
+        )
+      },
+      {
+        path: 'register',
+        element: (
+          <UnAuthOnlyRoute><Logistration /></UnAuthOnlyRoute>
+        ),
+      },
+      {
+        path: 'reset',
+        element: (
+          <UnAuthOnlyRoute><ForgotPasswordPage /></UnAuthOnlyRoute>
+        ),
+      },
+      {
+        path: 'password_reset_confirm/:token',
+        element: (
+          <ResetPasswordPage />
+        ),
+      },
+      {
+        path: 'welcome',
+        element: (
+          <ProgressiveProfiling />
+        ),
+      },
+      {
+        path: 'notfound',
+        element: (
+          <NotFoundPage />
+        ),
+      },
+      {
+        path: '*',
+        element: (
+          <Navigate replace to={PAGE_NOT_FOUND} />
+        ),
+      }
```

Comment:
Probably not a good idea having an app with a catch-all.  This should be handled by the shell.  (Thanks @bradenmacdonald for catching this.)